src/validation/errors/MatchFieldError.ts

```diff
@@ -0,0 +1,6 @@
+export class MatchFieldError extends Error {
+  constructor(message?: string) {
+    super(message ?? `Value does not match pattern`);
```

Comment:
instead of ??, I think || works better, as it is in src/domain/errors/UnexpectedError.ts